ledger/src/bigtable_upload.rs

```diff
@@ -51,21 +51,24 @@ pub async fn upload_confirmed_blocks(
 ) -> Result<Slot, Box<dyn std::error::Error>> {
     let mut measure = Measure::start("entire upload");
 
-    info!("Loading ledger slots starting at {}...", starting_slot);
+    info!(
+        "Loading ledger slots from {} to {}",
+        starting_slot, ending_slot
+    );
     let blockstore_slots: Vec<_> = blockstore
         .rooted_slot_iterator(starting_slot)
         .map_err(|err| {
             format!("Failed to load entries starting from slot {starting_slot}: {err:?}")
         })?
-        .map_while(|slot| (slot <= ending_slot).then_some(slot))
+        .take_while(|slot| *slot <= ending_slot)
         .collect();
 
     if blockstore_slots.is_empty() {
         return Err(format!("Ledger has no slots from {starting_slot} to {ending_slot:?}").into());
     }
 
-    let first_blockstore_slot = blockstore_slots.first().unwrap();
-    let last_blockstore_slot = blockstore_slots.last().unwrap();
+    let first_blockstore_slot = *blockstore_slots.first().unwrap();
+    let last_blockstore_slot = *blockstore_slots.last().unwrap();
```

Comment:
de-ref so we can drop the `Vec` with `into_iter()` instead of having to use `.cloned()`